contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,866 @@
+        return _validators[operatorAddress].commission;
+    }
+
+    function getValidatorWithVotingPower(
+        uint256 offset,
+        uint256 limit
+    ) external view returns (address[] memory consensusAddrs, uint256[] memory votingPowers, uint256 totalLength) {
+        limit = limit > 100 ? 100 : limit;
+        totalLength = _validatorSet.length();
+        if (offset >= totalLength) {
+            return (consensusAddrs, votingPowers, totalLength);
+        }
+
+        uint256 count = (totalLength - offset) > limit ? limit : (totalLength - offset);
+        consensusAddrs = new address[](count);
+        votingPowers = new uint256[](count);
+        for (uint256 i; i < count; ++i) {
+            address operatorAddress = _validatorSet.at(offset + i);
+            Validator memory valInfo = _validators[operatorAddress];
+            consensusAddrs[i] = valInfo.consensusAddress;
+            address pool = valInfo.poolModule;
+            votingPowers[i] = IStakePool(pool).totalPooledBNB();
+        }
+    }
+
+    function getOperatorAddressByVoteAddress(bytes calldata voteAddress) external view returns (address) {
+        return _voteToOperator[voteAddress];
+    }
+
+    function getOperatorAddressByConsensusAddress(address consensusAddress) external view returns (address) {
+        return _consensusToOperator[consensusAddress];
+    }
+
+    function getSlashRecord(
+        address operatorAddress,
+        uint256 height,
+        SlashType slashType
+    ) external view returns (uint256 slashAmount, uint256 slashHeight, uint256 jailUntil) {
+        bytes32 slashKey = _getSlashKey(operatorAddress, height, slashType);
+        SlashRecord memory record = _slashRecords[slashKey];
+        slashAmount = record.slashAmount;
+        slashHeight = record.slashHeight;
+        jailUntil = record.jailUntil;
+    }
+
+    /*----------------- internal functions -----------------*/
+    function _compareStrings(string memory a, string memory b) internal pure returns (bool) {
+        return (keccak256(abi.encodePacked((a))) == keccak256(abi.encodePacked((b))));
+    }
+
+    function _bytesToUint256(uint256 _offset, bytes memory _input) internal pure returns (uint256 _output) {
+        assembly {
+            _output := mload(add(_input, _offset))
+        }
+    }
+
+    function _bytesToAddress(uint256 _offset, bytes memory _input) internal pure returns (address _output) {
+        assembly {
+            _output := mload(add(_input, _offset))
+        }
+    }
+
+    function _getSlashKey(
+        address operatorAddress,
+        uint256 height,
+        SlashType slashType
+    ) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(operatorAddress, height, slashType));
+    }
+
+    function _checkMoniker(string memory moniker) internal pure returns (bool) {
+        bytes memory bz = bytes(moniker);
+
+        // 1. moniker length should be between 1 and 9
```

Comment:
==>`moniker length should be between 3 and 9`,  1 is too small

Reply:
Done